packages/passport/sdk/src/zkEvm/zkEvmProvider.ts

```diff
@@ -332,6 +334,24 @@ export class ZkEvmProvider implements Provider {
             const ethSigner = await this.#getSigner();
             flow.addEvent('endGetSigner');
 
+            if (this.#config.forceScwDeployBeforeMessageSignature) {
+              // Check if the smart contract wallet has been deployed
+              const nonce = await getNonce(this.#rpcProvider, zkEvmAddress);
+              if (!nonce.gt(0)) {
+                // If the smart contract wallet has not been deployed,
+                // submit a transaction before signing the message
+                return await sendDeployTransactionAndPersonalSign({
+                  params: request.params || [],
+                  ethSigner,
+                  zkEvmAddress,
+                  rpcProvider: this.#rpcProvider,
+                  guardianClient: this.#guardianClient,
+                  relayerClient: this.#relayerClient,
+                  flow,
+                });
+              }
+            }
```

Comment:
I think we should implement this functionality for EIP-712 signatures as well.

From a user's perspective, I don't believe that there is any good argument for this functionality to only exist for EIP-191 signatures, and by doing this check for EIP-191 signatures and not EIP-712 signatures, we're creating another quirk that consumers need to be aware of